asttokens/mark_tokens.py

```diff
@@ -61,14 +68,15 @@ def _visit_before_children(self, node, parent_token):
     return (token or parent_token, token)
 
   def _visit_after_children(self, node, parent_token, token):
+    # type: (NodeNG, Optional[util.Token], Optional[util.Token]) -> None
```

Comment:
There's still lots of `NodeNG` here and in `asttokens.py` that should really be `AstNode`. Note that the library adds several attributes like `first_token` to the nodes, you'll have to find a way to deal with that.

Reply:
I've replaced them with AstNode, and fixed the definition of that to use an `EnhancedAST` types marker class that basically just goes "we promise ASTs have some extra fields"